src/app/GitCommands/Remotes/ConfigFileRemoteSettingsManager.cs

```diff
@@ -147,7 +147,25 @@ bool IsSettingForBranch(string setting, string branchName)
                                    .Select(t => new GitRef(module, null, t[1]))
                                    .FirstOrDefault(h => h.IsHead);
 
-            return remoteHead?.Name;
+            if (remoteHead is not null)
+            {
+                return remoteHead.Name;
+            }
+
+            bool IsSettingForAnyBranch(string setting)
+            {
+                GitRef head = new(module, null, setting);
+                return head.IsHead && head.Name == "*";
+            }
+
+            GitRef remoteAnyHead = remote.Push
+                                   .Select(s => s.Split(Delimiters.Colon))
+                                   .Where(t => t.Length == 2)
+                                   .Where(t => IsSettingForAnyBranch(t[0]))
+                                   .Select(t => new GitRef(module, null, t[1].Replace("*", branch)))
+                                   .FirstOrDefault(h => h.IsHead);
```

Comment:
I would rather join this check and transformation into one function which also handles the special case "wildcard branch" without duplicating the LINQ chain from above.
`GitRef? GetRemoteHead(string pushSetting)`

```suggestion
            GitRef? remoteAnyHead = remote.Push
                                    .Select(GetRemoteHead)
                                    .FirstOrDefault(h => h.IsHead);
```

Reply:
I'm not sure how it's supposed to behave if you have _both_. Does first win, or more specific win?

Reply:
➡️ Did not change this for now, but I can see how it could be improved in the future if we decided to do so